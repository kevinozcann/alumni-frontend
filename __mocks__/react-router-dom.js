import React from 'react';

const rrd = jest.genMockFromModule('react-router-dom');

rrd.BrowserRouter = ({ children }) => <div>{children}</div>;

module.exports = rrd;
