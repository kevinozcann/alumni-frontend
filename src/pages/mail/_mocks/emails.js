export const emails = [
  {
    id: 6153,
    email: {
      id: 6074,
      subject: 'testing',
      msgBody: '<p>testttt</p>',
      sender: {
        uuid: 'simsek',
        email: 'simsek.mert@smartclass.com.tr',
        picture: 'https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg',
        roles: ['ROLE_USER', 'ROLE_ADMIN'],
        accessToken: 'abcdxyz',
        fullName: '\u015eim\u015fek Mert',
        isMe: true
      },
      sentAt: '2022-01-14T02:56:22+00:00'
    },
    user: {
      uuid: 'simsek',
      email: 'simsek.mert@smartclass.com.tr',
      picture: 'https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg',
      roles: ['ROLE_USER', 'ROLE_ADMIN'],
      accessToken: 'abcdxyz',
      fullName: '\u015eim\u015fek Mert',
      isMe: true
    },
    isSender: true,
    isTo: false,
    isCc: false,
    isBcc: false,
    isRead: true,
    isStarred: false,
    isTrashed: false,
    isDraft: false
  },
  {
    id: 6154,
    email: {
      id: 6074,
      subject: 'testing',
      msgBody: '<p>testttt</p>',
      sender: {
        uuid: 'simsek',
        email: 'simsek.mert@smartclass.com.tr',
        picture: 'https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg',
        roles: ['ROLE_USER', 'ROLE_ADMIN'],
        accessToken: 'abcdxyz',
        fullName: '\u015eim\u015fek Mert',
        isMe: true
      },
      sentAt: '2022-01-14T02:56:22+00:00'
    },
    user: {
      uuid: 'simsek',
      email: 'simsek.mert@smartclass.com.tr',
      picture: 'https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg',
      roles: ['ROLE_USER', 'ROLE_ADMIN'],
      accessToken: 'abcdxyz',
      fullName: '\u015eim\u015fek Mert',
      isMe: true
    },
    isSender: false,
    isTo: true,
    isCc: false,
    isBcc: false,
    isRead: false,
    isStarred: true,
    isTrashed: false,
    isDraft: false
  },
  {
    id: 6152,
    email: {
      id: 6073,
      subject: 'testing queue',
      msgBody: '<p>tssssss</p>',
      sender: {
        uuid: 'system',
        email: 'no-reply@smartclass.biz',
        picture: '',
        fullName: 'System ',
        isMe: false
      },
      sentAt: '2022-01-14T02:25:14+00:00'
    },
    user: {
      uuid: 'simsek',
      email: 'simsek.mert@smartclass.com.tr',
      picture: 'https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg',
      roles: ['ROLE_USER', 'ROLE_ADMIN'],
      accessToken: 'abcdxyz',
      fullName: '\u015eim\u015fek Mert',
      isMe: true
    },
    isSender: false,
    isTo: true,
    isCc: false,
    isBcc: false,
    isRead: true,
    isStarred: false,
    isTrashed: false,
    isDraft: false
  },
  {
    id: 5980,
    email: {
      id: 5800,
      attachment: '',
      subject: 'good morning',
      msgBody: 'thank you',
      sender: {
        uuid: 'ugur',
        email: 'ugur.tekbacak@smartclass.com.tr',
        picture:
          'https://fs-demo.smartclass.tech/files/userfiles/ugur/air-force-1-shadow-ayakkab%C4%B1s%C4%B1-r48jh3.jpg',
        fullName: 'U\u011fur Tekbacak',
        isMe: false
      },
      sentAt: '2021-11-18T13:11:20+00:00'
    },
    user: {
      uuid: 'simsek',
      email: 'simsek.mert@smartclass.com.tr',
      picture: 'https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg',
      roles: ['ROLE_USER', 'ROLE_ADMIN'],
      accessToken: 'abcdxyz',
      fullName: '\u015eim\u015fek Mert',
      isMe: true
    },
    isSender: false,
    isTo: true,
    isCc: false,
    isBcc: false,
    isRead: true,
    isStarred: false,
    isTrashed: false,
    isDraft: false
  },
  {
    id: 5977,
    email: {
      id: 5799,
      subject: 'good morning',
      msgBody: '<p>how are you doing</p>',
      sender: {
        uuid: 'simsek',
        email: 'simsek.mert@smartclass.com.tr',
        picture: 'https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg',
        roles: ['ROLE_USER', 'ROLE_ADMIN'],
        accessToken: 'abcdxyz',
        fullName: '\u015eim\u015fek Mert',
        isMe: true
      },
      sentAt: '2021-11-18T13:10:50+00:00'
    },
    user: {
      uuid: 'simsek',
      email: 'simsek.mert@smartclass.com.tr',
      picture: 'https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg',
      roles: ['ROLE_USER', 'ROLE_ADMIN'],
      accessToken: 'abcdxyz',
      fullName: '\u015eim\u015fek Mert',
      isMe: true
    },
    isSender: true,
    isTo: false,
    isCc: false,
    isBcc: false,
    isRead: true,
    isStarred: false,
    isTrashed: false,
    isDraft: false
  },
  {
    id: 5879,
    email: {
      id: 5695,
      subject: 'testing again',
      msgBody: '<p>testing again and again</p>',
      sender: {
        uuid: 'simsek',
        email: 'simsek.mert@smartclass.com.tr',
        picture: 'https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg',
        roles: ['ROLE_USER', 'ROLE_ADMIN'],
        accessToken: 'abcdxyz',
        fullName: '\u015eim\u015fek Mert',
        isMe: true
      },
      sentAt: '2021-10-08T06:43:16+00:00'
    },
    user: {
      uuid: 'simsek',
      email: 'simsek.mert@smartclass.com.tr',
      picture: 'https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg',
      roles: ['ROLE_USER', 'ROLE_ADMIN'],
      accessToken: 'abcdxyz',
      fullName: '\u015eim\u015fek Mert',
      isMe: true
    },
    isSender: false,
    isTo: true,
    isCc: false,
    isBcc: false,
    isRead: true,
    isStarred: false,
    isTrashed: false,
    isDraft: false
  }
];

export default emails;
