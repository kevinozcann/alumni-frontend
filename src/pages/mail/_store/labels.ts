import { IMailLabel } from 'pages/mail/mail-types';

const labels: IMailLabel[] = [
  {
    id: 'inbox',
    type: 'system_label',
    name: 'inbox',
    unreadCount: 0,
    totalCount: 0
  },
  {
    id: 'starred',
    type: 'system_label',
    name: 'starred',
    unreadCount: 0,
    totalCount: 0
  },
  {
    id: 'sent',
    type: 'system_label',
    name: 'sent',
    unreadCount: 0,
    totalCount: 0
  },
  {
    id: 'drafts',
    type: 'system_label',
    name: 'drafts',
    unreadCount: 0,
    totalCount: 0
  },
  {
    id: 'trash',
    type: 'system_label',
    name: 'trash',
    unreadCount: 0,
    totalCount: 0
  }
  // {
  //   id: '5e892628d4bc60b4514d5d36',
  //   type: 'custom_label',
  //   name: 'Work',
  //   unreadCount: 1,
  //   totalCount: 1
  // },
  // {
  //   id: '5e8926820cf9ec6c834114ec',
  //   type: 'custom_label',
  //   name: 'Business',
  //   unreadCount: 0,
  //   totalCount: 2
  // }
];

export default labels;
