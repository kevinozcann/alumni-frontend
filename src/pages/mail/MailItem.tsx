import React from 'react';
import { Link as RouterLink, useParams } from 'react-router-dom';
// import { useIntl } from 'react-intl';
import moment from 'moment';
import {
  Avatar,
  Box,
  Checkbox,
  Chip,
  IconButton,
  Tooltip,
  Typography,
  useTheme
} from '@mui/material';
import { amber } from '@mui/material/colors';
import { Star, StarBorder } from '@mui/icons-material';
import ShortcutIcon from '@mui/icons-material/Shortcut';
import getInitials from 'utils/getInitials';

import { IMail, IMailLabel, IUserMail } from './mail-types';
import useTranslation from 'hooks/useTranslation';

interface MailItemProps {
  handleStarClick: (mailId: number, starred: boolean) => void;
  handleForwardClick: (email: IMail) => void;
  labels: IMailLabel[];
  mail: IUserMail;
  onDeselect?: () => void;
  onSelect?: () => void;
  selected: boolean;
}

const getTo = (section: string, label: string, mailId: number) => {
  const baseUrl = '/mail';

  if (section) {
    return `${baseUrl}/${section}/${mailId}`;
  }

  if (label) {
    return `${baseUrl}/label/${label}/${mailId}`;
  }

  return baseUrl;
};

const MailItem = (props: MailItemProps) => {
  const { handleStarClick, handleForwardClick, mail, onDeselect, onSelect, selected } = props;
  const theme = useTheme();
  const { section, label } = useParams();
  const {
    id: mailId,
    email: { subject, sender, sentAt, emailToUser },
    isTo,
    isCc,
    isBcc,
    isRead,
    isStarred
  } = mail;
  const [starred, setStarred] = React.useState(isStarred);
  const intl = useTranslation();

  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    const { checked } = event.target;

    if (checked && onSelect) {
      onSelect();
    }

    if (!checked && onDeselect) {
      onDeselect();
    }
  };

  const handleStarToggle = (): void => {
    setStarred(!starred);
    handleStarClick(mailId, !starred);
  };

  const draftValues: IMail = {
    subject: mail?.email?.subject,
    msgBody: mail?.email?.msgBody
  };

  const handleForwardToggle = (): void => {
    handleForwardClick(draftValues);
  };

  const to = getTo(section, label, mailId);

  return (
    <Box
      sx={{
        alignItems: 'center',
        backgroundColor: 'background.paper',
        borderBottom: (theme) => ` 1px solid ${theme.palette.divider}`,
        display: 'flex',
        p: 2,
        ...(!isRead && {
          position: 'relative',
          '&:before': {
            backgroundColor: 'error.main',
            content: '" "',
            height: '100%',
            left: 0,
            position: 'absolute',
            top: 0,
            width: 4
          },
          '& $name, & $subject': {
            fontWeight: 600
          }
        }),
        ...(selected && {
          backgroundColor: 'action.selected'
        }),
        '&:hover': {
          backgroundColor: 'action.hover'
        }
      }}
    >
      <Box sx={{ display: { xs: 'none', sm: 'block' } }}>
        <Box
          sx={{
            alignItems: 'center',
            display: 'flex',
            mr: 1
          }}
        >
          <Checkbox checked={selected} color='primary' onChange={handleCheckboxChange} />
          <Tooltip title='Starred'>
            <IconButton onClick={handleStarToggle}>
              {mail.isStarred ? (
                <Star fontSize='small' sx={{ color: amber[400] }} />
              ) : (
                <StarBorder fontSize='small' />
              )}
            </IconButton>
          </Tooltip>

          <Tooltip title={intl.formatMessage({ id: 'forward' })}>
            <IconButton onClick={handleForwardToggle}>
              <ShortcutIcon fontSize='small' />
            </IconButton>
          </Tooltip>
          {/* <Tooltip title='Important'>
            <IconButton onClick={handleImportantToggle}>
              {mail.isImportant ? (
                <LabelImportantIcon fontSize='small' sx={{ color: amber[400] }} />
              ) : (
                <LabelImportantIcon fontSize='small' />
              )}
            </IconButton>
          </Tooltip> */}
        </Box>
      </Box>
      <Box
        component={RouterLink}
        to={to}
        sx={{
          cursor: 'pointer',
          display: 'flex',
          flexGrow: 1,
          minWidth: '1px',
          textDecoration: 'none'
        }}
      >
        {section == 'sent' ? (
          <Avatar
            sx={{ backgroundColor: 'primary.main' }}
            alt={emailToUser?.fullName}
            src={emailToUser?.picture}
          >
            {getInitials(emailToUser?.fullName)}
          </Avatar>
        ) : (
          <Avatar
            sx={{ backgroundColor: 'primary.main' }}
            alt={sender?.fullName || sender?.name}
            src={sender?.picture}
          >
            {getInitials(sender?.fullName || sender?.name)}
          </Avatar>
        )}

        <Box
          sx={{
            alignItems: {
              md: 'center'
            },
            display: {
              md: 'flex'
            },
            flexGrow: {
              md: 1
            },
            minWidth: '1px',
            ml: 1
          }}
        >
          <Typography
            component='div'
            color={theme.palette.text.secondary}
            sx={{
              flexBasis: {
                md: 180
              },
              minWidth: {
                md: 180
              },
              ...(mail.isRead && {
                fontWeight: 600
              })
            }}
          >
            {section == 'sent' ? emailToUser?.fullName : sender?.fullName}
            {isTo && (
              <Chip
                sx={{
                  fontSize: 7,
                  height: 11,
                  ml: 1,
                  '& span': {
                    padding: 1
                  }
                }}
                label='TO'
              />
            )}
            {isCc && (
              <Chip
                sx={{
                  fontSize: 7,
                  height: 11,
                  ml: 1,
                  '& span': {
                    padding: 1
                  }
                }}
                label='CC'
              />
            )}
            {isBcc && (
              <Chip
                sx={{
                  fontSize: 7,
                  height: 11,
                  ml: 1,
                  '& span': {
                    padding: 1
                  }
                }}
                label='BCC'
              />
            )}
          </Typography>
          <Typography
            color='textSecondary'
            sx={{
              maxWidth: 400,
              mr: 2,
              overflow: 'hidden',
              textOverflow: 'ellipsis',
              whiteSpace: 'nowrap',
              ...(mail.isRead && {
                fontWeight: 600
              })
            }}
            variant='body2'
          >
            {subject}
          </Typography>
          <Box sx={{ display: { md: 'block', xs: 'none' } }}>
            <Typography
              color='textSecondary'
              sx={{
                flexGrow: 1,
                marginRight: 'auto',
                maxWidth: 800,
                overflow: 'hidden',
                textOverflow: 'ellipsis',
                whiteSpace: 'nowrap'
              }}
              variant='body2'
            >
              {/* -{msgBody} */}
            </Typography>
            {/* {mail.labelIds.length > 0 && (
              <Box
                sx={{
                  display: 'flex',
                  mx: 2
                }}>
                {mail.labelIds.map((labelId: number) => {
                  const label = labels.find((_label) => parseInt(_label.id) === labelId);

                  if (!label) {
                    return null;
                  }

                  return (
                    // <Label key={label.id} sx={{ backgroundColor: label.color }}>
                    <Label key={label.id}>{label.name}</Label>
                  );
                })}
              </Box>
            )} */}
          </Box>
          <Typography color='textSecondary' noWrap variant='caption'>
            {moment(sentAt).format('llll')}
          </Typography>
        </Box>
      </Box>
    </Box>
  );
};

export default MailItem;
