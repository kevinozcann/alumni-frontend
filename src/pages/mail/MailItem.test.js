import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import MailItem from './MailItem';
import labels from './_store/labels';
import emails from './_mocks/emails';

describe('MailItem component snapshot', () => {
  it('should render as expected', () => {
    const email = emails[0];

    const tree = renderer
      .create(
        <MemoryRouter>
          <MailItem
            handleStarClick={() => console.log('starred')}
            labels={labels}
            mail={email}
            selected={false}
          />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
