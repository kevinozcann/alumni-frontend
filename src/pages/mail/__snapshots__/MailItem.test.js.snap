// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailItem component snapshot should render as expected 1`] = `
<div
  className="MuiBox-root css-l9ygfb"
>
  <div
    className="MuiBox-root css-130f8nx"
  >
    <div
      className="MuiBox-root css-vkm8w4"
    >
      <span
        className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <button
        aria-label="Starred"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="StarBorderIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <a
    className="MuiBox-root css-1mswlbp"
    href="/mail"
    onClick={[Function]}
    sx={
      Object {
        "cursor": "pointer",
        "display": "flex",
        "flexGrow": 1,
        "minWidth": "1px",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular css-2k1ilz-MuiAvatar-root"
    >
      <img
        alt="Şimşek Mert"
        className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="https://fs-demo.smartclass.tech/files/userfiles/simsek/mteacher.jpg"
      />
    </div>
    <div
      className="MuiBox-root css-1tkjsiq"
    >
      <div
        className="MuiTypography-root MuiTypography-body1 css-oat79r-MuiTypography-root"
      >
        Şimşek Mert
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 css-og44s1-MuiTypography-root"
      >
        testing
      </p>
      <div
        className="MuiBox-root css-13tqxrv"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-143aoc4-MuiTypography-root"
        />
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-z0slbs-MuiTypography-root"
      >
        Fri, Jan 14, 2022 5:56 AM
      </span>
    </div>
  </a>
</div>
`;
